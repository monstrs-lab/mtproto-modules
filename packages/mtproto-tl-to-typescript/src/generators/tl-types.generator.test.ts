import { tmpdir }               from 'node:os'
import { join }                 from 'node:path'

import { describe }             from '@jest/globals'
import { expect }               from '@jest/globals'
import { it }                   from '@jest/globals'
import { Project }              from 'ts-morph'
import { ScriptTarget }         from 'ts-morph'
import { ModuleKind }           from 'ts-morph'
import { ModuleResolutionKind } from 'ts-morph'

import { TLTypesGenerator }     from './tl-types.generator.js'

describe('mtproto tl to typescript', () => {
  describe('tl types generator', () => {
    it('check generate types', async () => {
      const project = new Project({
        compilerOptions: {
          outDir: tmpdir(),
          module: ModuleKind.ES2022,
          target: ScriptTarget.ES2022,
          moduleResolution: ModuleResolutionKind.Bundler,
        },
      })

      project.createSourceFile(join(tmpdir(), 'test1.ts')).addClass({
        isExported: true,
        name: 'Test1',
        properties: [
          {
            name: 'type',
            type: 'string',
            initializer: 'Test',
          },
        ],
      })

      project.createSourceFile(join(tmpdir(), 'test2.ts')).addClass({
        isExported: true,
        name: 'Test2',
        properties: [
          {
            name: 'type',
            type: 'string',
            initializer: 'Test',
          },
        ],
      })

      new TLTypesGenerator(project).generate()

      expect(project.getSourceFile(join(tmpdir(), 'test.ts'))?.getFullText()).toMatchSnapshot()
    })
  })
})
