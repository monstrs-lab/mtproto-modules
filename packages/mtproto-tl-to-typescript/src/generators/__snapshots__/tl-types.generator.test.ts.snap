// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mtproto tl to typescript tl types generator check generate types 1`] = `
"import type { Test1 } from "./test1.js";
import type { Test2 } from "./test2.js";

export type TypeTest = Test1 | Test2;
"
`;
